src/Bicep.Core/Semantics/YamlObjectParser.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.TypeSystem;
+using Newtonsoft.Json.Linq;
+using SharpYaml.Serialization;
+
+namespace Bicep.Core.Semantics
+{
+    public class YamlObjectParser : ObjectParser
+    {
+        override public JToken ExtractTokenFromObject(string fileContent)
+            => JToken.FromObject(new Serializer().Deserialize(fileContent)!);
```

Comment:
`!` is suppressing C#'s nullable analysis, meaning there's a chance this could throw an exception. Instead, please make sure you're handling `null` properly - e.g. something like:

```
public override `JToken? TryExtractTokenFromObject(string fileContent)
    => new Serializer().Deserialize(fileContent) is {} deserialized ? JToken.FromObject(deserialized) : null;
```

Reply:
Please make sure you've got tests to cover these functions being called with unparseable YAML btw - to ensure we handle gracefully rather than throwing an exception.

Reply:
we added tests for both yaml and json.